File: TCAlertView/ViewController.m
//
//  ViewController.m
//  TCAlertView
//  Copyright (c) 2016 Technocracker. All rights reserved.
//

#import "ViewController.h"
#import "TCAlertView.h"

@interface ViewController ()<UITableViewDataSource,UITableViewDelegate>
{
    NSArray *arrFeature;
    UITableView *tblList;
}
@end

@implementation ViewController

- (void)viewDidLoad
{
    [super viewDidLoad];
    arrFeature = [[NSArray alloc] initWithObjects:@"Toast Message Without Button",@"Toast Message With Button",@"One Button",@"Two Button",@"Multiple Button", nil];
    CGSize screen = [UIScreen mainScreen].bounds.size;
    
    tblList = [[UITableView alloc] initWithFrame:CGRectMake(0,0,screen.width,screen.height)];
    tblList.dataSource = self;
    tblList.delegate = self;
    [self.view addSubview:tblList];
    
}
- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section
{
    return arrFeature.count;
}
- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath
{
    NSString *strIdentifier=[NSString stringWithFormat:@"Cell%ld",(long)indexPath.row];
    
    UITableViewCell *cell=(UITableViewCell *) [tableView dequeueReusableCellWithIdentifier:strIdentifier];
    if (cell == nil)
    {
        cell=[[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:strIdentifier];
    }
    cell.textLabel.text = arrFeature[indexPath.row];
    return cell;
}
-(void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath *)indexPath
{
    switch (indexPath.row)
    {
        case 0:
            [[TCAlertView sharedInstance] show:@"Toast Message" message:@"This toast message" buttonTitles:@[] interval:3.0f btnClick:^(int buttonIndex) {
                NSLog(@"button Index %d",buttonIndex);
            }];
            break;
        case 1:
            [[TCAlertView sharedInstance] show:@"Toast Message" message:@"This toast message" buttonTitles:@[@"OK",@"Cancel"] interval:3.0f btnClick:^(int buttonIndex) {
                NSLog(@"button Index %d",buttonIndex);
            }];
            break;
        case 2:
            [[TCAlertView sharedInstance] show:@"Title" message:@"TCAlertView Message With One Button" buttonTitles:@[@"Ok"] btnClick:^(int buttonIndex) {
                NSLog(@"button Index %d",buttonIndex);
            }];
            break;
            
        case 3:
            [[TCAlertView sharedInstance] show:@"Title" message:@"TCAlertView Message With Two Button" buttonTitles:@[@"Ok",@"Cancel"] btnClick:^(int buttonIndex) {
                NSLog(@"button Index %d",buttonIndex);
            }];
            break;
        case 4:
            [[TCAlertView sharedInstance] show:@"Title" message:@"TCAlertView Message With Multiple Button" buttonTitles:@[@"One",@"Two",@"Three",@"Four",@"Five"] btnClick:^(int buttonIndex) {
                NSLog(@"button Index %d",buttonIndex);
            }];
            break;
            
        default:
            break;
    }
}
- (void)didReceiveMemoryWarning
{
    [super didReceiveMemoryWarning];
}

@end

File: Swift Exmaple/TCAlertViewSwift/TCAlertViewSwift/ViewController.swift
//
//  ViewController.swift
//  TCAlertViewSwift
//
//  Created by Technocracker on 2/23/16.
//  Copyright (c) 2016 Technocracker. All rights reserved.
//

import UIKit

class ViewController: UIViewController,UITableViewDataSource,UITableViewDelegate
{

    var tblFeature:UITableView!
    var arrFeature:[String]!
    override func viewDidLoad()
    {
        super.viewDidLoad()
        
        arrFeature = ["Toast Message Without Button","Toast Message With Button","One Button","Two Button","Multiple Button"]
        
        let Screen = UIScreen.mainScreen().bounds
        tblFeature = UITableView(frame:CGRectMake(0,10,Screen.width,Screen.height))
        tblFeature.dataSource = self
        tblFeature.delegate = self
        self.view.addSubview(tblFeature)
    }
    func tableView(tableView: UITableView, numberOfRowsInSection section: Int) -> Int
    {
        return arrFeature.count
    }
    func tableView(tableView: UITableView, cellForRowAtIndexPath indexPath: NSIndexPath) -> UITableViewCell
    {
        var cell = tableView.dequeueReusableCellWithIdentifier("cell\(indexPath.row)") as? UITableViewCell
        if cell == nil
        {
           cell = UITableViewCell(style: UITableViewCellStyle.Default, reuseIdentifier:"cell\(indexPath.row)")
        }
        cell?.textLabel?.text = arrFeature[indexPath.row]
        return cell!
    }
    func tableView(tableView: UITableView, didSelectRowAtIndexPath indexPath: NSIndexPath)
    {
        switch indexPath.row
        {
            case 0:
                TCAlertView.sharedInstance().show("Alert", message:"This is Toast Message", buttonTitles:[], interval:3, btnClick: { (buttonIndex) -> Void in
                    print(buttonIndex)
                })
                break
            case 1:
                TCAlertView.sharedInstance().show("Alert", message:"This is Toast Message", buttonTitles:["Ok","Cancel"], interval:3, btnClick: { (buttonIndex) -> Void in
                    print(buttonIndex)
                })
                break
            case 2:
                TCAlertView.sharedInstance().show("Alert", message:"Toast Message With One Button", buttonTitles:["Ok"], btnClick: { (buttonIndex) -> Void in
                    print(buttonIndex)
                })
                break
            case 3:
                TCAlertView.sharedInstance().show("Alert", message:"Toast Message With Two Button", buttonTitles:["Ok","Cancel"], btnClick: { (buttonIndex) -> Void in
                    print(buttonIndex)
                })
                break
            case 4:
                TCAlertView.sharedInstance().show("Alert", message:"Toast Message With Multiple Button", buttonTitles:["1","2","3","4","5","6"], btnClick: { (buttonIndex) -> Void in
                    print(buttonIndex)
                })
                break
            default:
                break
                
        }
        
    }
    override func didReceiveMemoryWarning()
    {
        super.didReceiveMemoryWarning()
    }


}

